File: src/application/ApplicationCommandInitializer.java
package application;

import commands.client_side.*;
import commands.client_side.core.Command;
import lib.collections.ImmutablePair;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class ApplicationCommandInitializer {
    private static ImmutablePair<String, Command> getCommandPair(String localeName, Command command) {
        return new ImmutablePair<>(localeName, command);
    }

    public static List<ImmutablePair<String, Command>> getCommand(Application application) {
        return new ArrayList<>(
                Arrays.asList(
                        getCommandPair(
                                "command.help",
                                new HelpCommand(
                                        application::helpCommandCallback,
                                        application
                                )
                        ),
                        getCommandPair(
                                "command.info",
                                new InfoCommand(
                                        application::infoCommandCallback,
                                        application.getOrganizationManager()
                                )
                        ),
                        getCommandPair(
                                "command.show",
                                new ShowCommand(
                                        application::showCommandCallback,
                                        application.getOrganizationManager()
                                )
                        ),
                        getCommandPair(
                                "command.add",
                                new AddCommand(
                                        application::addCommandCallback,
                                        application,
                                        application.getOrganizationManager()
                                )
                        ),
                        getCommandPair(
                                "command.update",
                                new UpdateCommand(
                                        application::modifyOrganizationCommandCallback,
                                        application,
                                        application.getOrganizationManager()
                                )
                        ),
                        getCommandPair(
                                "command.remove_by_id",
                                new RemoveByIdCommand(
                                        application::removeCommandCallback,
                                        application.getOrganizationManager()
                                )
                        ),
                        getCommandPair(
                                "command.clear",
                                new ClearCommand(
                                        application::clearCommandCallback,
                                        application.getOrganizationManager()
                                )
                        ),
                        getCommandPair(
                                "command.save",
                                new SaveCommand(
                                        application::saveCommandCallback,
                                        application.getOrganizationManager()
                                )
                        ),
                        getCommandPair(
                                "command.read",
                                new ReadCommand(
                                        application::readCommandCallback,
                                        application.getOrganizationManager()
                                )
                        ),
                        getCommandPair(
                                "command.execute_script",
                                new ExecuteScriptCommand(
                                        application::executeScriptCommandCallback,
                                        application
                                )
                        ),
                        getCommandPair(
                                "command.exit",
                                new ExitCommand(
                                        application::exitCommandCallback,
                                        application
                                )
                        ),
                        getCommandPair(
                                "command.remove_head",
                                new RemoveHeadCommand(
                                        application::removeHeadCommandCallback,
                                        application.getOrganizationManager()
                                )
                        ),
                        getCommandPair(
                                "command.add_if_max",
                                new AddIfMaxCommand(
                                        application::addMaxCommandCallback,
                                        application,
                                        application.getOrganizationManager()
                                )
                        ),
                        getCommandPair(
                                "command.history",
                                new PrintHistoryCommand(
                                        application::showHistoryCommandCallback,
                                        application)
                        ),
                        getCommandPair(
                                "command.max_by_full_name",
                                new MaxByFullNameCommand(
                                        application::maxByFullNameCommandCallback,
                                        application.getOrganizationManager())
                        ),
                        getCommandPair(
                                "command.remove_all_by_postal_address",
                                new RemoveAllByPostalAddressCommand(
                                        application::removeAllByPostalAddressCommandCallback,
                                        application,
                                        application.getOrganizationManager()
                                )
                        ),
                        getCommandPair(
                                "command.sum_of_annual_turnover",
                                new SumOfAnnualTurnoverCommand(
                                        application::sumOfAnnualTurnoverCommandCallback,
                                        application.getOrganizationManager()
                                )
                        )
                )
        );
    }
}

File: src/application/ApplicationCommandInitializer.kt
package application

import commands.client_side.*
import commands.client_side.core.Command
import lib.collections.ImmutablePair
import network.client.DatabaseCommand

object ApplicationCommandInitializer {
    fun getCommand(application: Application): Map<DatabaseCommand, Command> {
        return mapOf(
            DatabaseCommand.HELP to HelpCommand(application),
            DatabaseCommand.INFO to InfoCommand(application.organizationManager),
            DatabaseCommand.SHOW to ShowCommand(application.organizationManager),
            DatabaseCommand.ADD to AddCommand(application, application.organizationManager),
            DatabaseCommand.UPDATE to UpdateCommand(application, application.organizationManager),
            DatabaseCommand.REMOVE_BY_ID to RemoveByIdCommand(application.organizationManager),
            DatabaseCommand.CLEAR to ClearCommand(application.organizationManager),
            DatabaseCommand.SAVE to SaveCommand(application.organizationManager),
            DatabaseCommand.READ to ReadCommand(application.organizationManager),
            DatabaseCommand.EXECUTE_SCRIPT to ExecuteScriptCommand(application),
            DatabaseCommand.EXIT to ExitCommand(application),
            DatabaseCommand.REMOVE_HEAD to RemoveHeadCommand(application.organizationManager),
            DatabaseCommand.ADD_IF_MAX to AddIfMaxCommand(application, application.organizationManager),
            DatabaseCommand.HISTORY to PrintHistoryCommand(application),
            DatabaseCommand.MAX_BY_FULL_NAME to MaxByFullNameCommand(application.organizationManager),
            DatabaseCommand.REMOVE_ALL_BY_POSTAL_ADDRESS to RemoveAllByPostalAddressCommand(
                application,
                application.organizationManager
            ),
            DatabaseCommand.SUM_OF_ANNUAL_TURNOVER to SumOfAnnualTurnoverCommand(application.organizationManager)
        )
    }
}
